src/core/shards/core.hpp

```diff
@@ -777,56 +778,59 @@ struct SetBase : public VariableBase {
     _key.warmup(context);
   }
 
-  ALWAYS_INLINE SHVar activate(SHContext *context, const SHVar &input) {
+  ALWAYS_INLINE void checkIfTableChanged() {
+    if (_tablePtr != _target->payload.tableValue.opaque) {
+      _tablePtr = _target->payload.tableValue.opaque;
+      _cell = nullptr;
+      SHLOG_TRACE("Table {} changed, clearing cell pointer", _name);
+    }
+  }
+};
+
+struct SetUpdateBase : public SetBase {
+  ALWAYS_INLINE SHVar activateTable(SHContext *context, const SHVar &input) {
+    checkIfTableChanged();
+
     if (likely(_cell != nullptr)) {
       cloneVar(*_cell, input);
       return *_cell;
     }
 
-    // if flagged exposed and not None, we got set already!
-    const auto shouldNotOverwrite = (_target->flags & SHVAR_FLAGS_EXPOSED) && _target->valueType != SHType::None;
+    if (_target->valueType != SHType::Table) {
+      if (_target->valueType != SHType::None)
+        destroyVar(*_target);
 
-    if (_isTable) {
-      if (_target->valueType != SHType::Table) {
-        if (_target->valueType != SHType::None)
-          destroyVar(*_target);
-
-        // Not initialized yet
-        _target->valueType = SHType::Table;
-        _target->payload.tableValue.api = &GetGlobals().TableInterface;
-        _target->payload.tableValue.opaque = new SHMap();
-      }
-
-      auto &kv = _key.get();
-      SHVar *vptr = _target->payload.tableValue.api->tableAt(_target->payload.tableValue, kv.payload.stringValue);
+      // Not initialized yet
+      _target->valueType = SHType::Table;
+      _target->payload.tableValue.api = &GetGlobals().TableInterface;
+      _target->payload.tableValue.opaque = new SHMap();
+    }
 
-      if (likely(!shouldNotOverwrite)) {
-        // Clone will try to recycle memory and such
-        cloneVar(*vptr, input);
-      }
+    auto &kv = _key.get();
+    SHVar *vptr = _target->payload.tableValue.api->tableAt(_target->payload.tableValue, kv.payload.stringValue);
 
-      // use fast cell from now
-      // if variable we cannot do this tho!
-      if (!_key.isVariable())
-        _cell = vptr;
+    // Clone will try to recycle memory and such
+    cloneVar(*vptr, input);
 
-      return *vptr;
-    } else {
-      if (likely(!shouldNotOverwrite)) {
-        // Clone will try to recycle memory and such
-        cloneVar(*_target, input);
-      }
+    // use fast cell from now
+    // if variable we cannot do this tho!
+    if (!_key.isVariable())
+      _cell = vptr;
 
-      // use fast cell from now
-      _cell = _target;
+    return *vptr;
+  }
 
-      return *_cell;
-    }
+  ALWAYS_INLINE SHVar activate(SHContext *context, const SHVar &input) {
+    // Clone will try to recycle memory and such
+    cloneVar(*_target, input);
+    return *_target;
   }
 };
 
-struct Set : public SetBase {
+struct Set : public SetUpdateBase {
   bool _exposed{false};
+  Shard *_self{};
+  entt::connection _onStartConnection{};
```

Comment:
What is this event for?

Reply:
Some lines under, but basically to avoid updating variables that rare sets when loading domain. (Will apply to exposed variables)